run.py

```diff
@@ -24,8 +24,7 @@
                                      get_git_hash, Logger
 from open_seq2seq.utils import train, infer, evaluate
 
-
-def main():
+def main(args, graph=None, namescope=None):
```

Comment:
I would suggest removing the calls to train/eval/infer from this function and rename it to something like create_model. That way it will be clear that this function will just create a corresponding model from the config and after if __name__ == '__main__': line we can call train/eval/infer on the created model. It will make the code more modular and logical, in my opinion.

Reply:
Also, it might be useful to split the function into subfunctions further, e.g. have a separate function for reading config, separate for checking parameters etc.

Reply:
And probably all these function should go inside the main open_seq2seq folder so that they come with all the rest of the code and run.py will be very simple as it should be.